pyhap/hap_server.py

```diff
@@ -737,8 +736,16 @@ def recv(self, buflen=1042, flags=0):
                 block_length_bytes = self.socket.recv(self.LENGTH_LENGTH)
                 if not block_length_bytes:
                     return result
-                # TODO: handle this
-                assert len(block_length_bytes) == self.LENGTH_LENGTH
+                if len(block_length_bytes) != self.LENGTH_LENGTH:
+                    # Handle a short read where we only get
+                    # one bytes.  This should be very rare, but it happens.
+                    block_length_bytes_remainder = self.socket.recv(
+                        self.LENGTH_LENGTH - len(block_length_bytes)
+                    )
+                    if not block_length_bytes_remainder:
+                        return result
```

Comment:
How about we change the way we read the length header like `block_length_bytes = self.socket.recv(self.LENGTH_LENGTH, flags=socket.MSG_WAITALL)`, forcing the socket to get us LENGTH_LENGTH before returning? This would simplify the logic a lot in exchange for crappy performance in the case where a block ends just at `LENGTH_LENGTH - 1` (which should seldom happen).

Reply:
Looks like that has broad cross OS support.  Good call. I'll rework and then test for a bit

Reply:
So far so good.  I'll keep testing over the rest of the day